src/VitessceGrid.js

```diff
@@ -36,106 +35,98 @@ function shallowEqual(objA, objB) {
     if (valueA !== valueB) {
       return false;
     }
-  }
+  });
 
   return true;
 }
 
-
-export default class VitessceGrid extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { readyComponentKeys: new Set() }; // eslint-disable-line react/no-unused-state
-    // This is being used and I'm not sure why there's a linting problem.
-  }
-
-  shouldComponentUpdate(nextProps) {
-    // We are ignoring state right now and only considering props.
-    // State is updated as components trigger onReady...
-    // but those events should not cause a re-render.
-    // React docs warn:
-    // > Do not rely on it to “prevent” a rendering, as this can lead to bugs.
-    // so there is probably a better approach.
-    return !shallowEqual(this.props, nextProps);
-  }
-
-  render() {
-    const {
-      layout, getComponent, padding, margin, draggableHandle,
-      reactGridLayoutProps, onAllReady, rowHeight,
-    } = this.props;
-
-    const ResponsiveGridLayout = WidthProvider(Responsive);
-
-    const {
-      cols, layouts, breakpoints, components,
-    } = resolveLayout(layout);
-
-    // Inline CSS is generally avoided, but this saves the end-user a little work,
-    // and prevents class names from getting out of sync.
-    const style = (
-      <style>
-        {`
+const VitessceGrid = (props) => {
+  const {
+    layout, getComponent, padding, margin, draggableHandle,
+    reactGridLayoutProps, onAllReady, rowHeight,
+  } = props;
+  const ResponsiveGridLayout = WidthProvider(Responsive);
+  const {
+    cols, layouts, breakpoints, components,
+  } = resolveLayout(layout);
+
+  const readyComponentKeys = new Set(); // only created when props change due to React.Memo
```

Comment:
Need to think more about whether to have this managed by `useState`. Now we allow each child to add to this set when ready, avoiding re-renders. 

Ideally we would like a way to add/remove components without re-rendering all components. Right now this is the case.